### PR TITLE
Refactoring of the Plugin class in BroadcastCacheUpdate

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface

It ended up touching many more files/docs/tests than I thought it would, so I've limited this PR to just handle it for the `BroadcastCacheUpdate` class. I'll do follow-up PRs for the other libraries, and use #282 to track all of them.

### CODE REVIEW

File: packages/sw-broadcast-cache-update/src/index.js [hunk 1]
```diff
@@ -57,13 +58,15 @@
...
+import BroadcastCacheUpdate from './lib/broadcast-cache-update';
+import BroadcastCacheUpdatePlugin from './lib/broadcast-cache-update-plugin';
 import broadcastUpdate from './lib/broadcast-update';
 import {cacheUpdatedMessageType} from './lib/constants';
 import responsesAreSame from './lib/responses-are-same';
 
 export {
-  Plugin,
+  BroadcastCacheUpdate,
+  BroadcastCacheUpdatePlugin,
```
Comment (gauntface) [inline packages/sw-broadcast-cache-update/src/index.js]: Nothing to block on but I'm not crazy about this naming `goog.broadcastUpdate.BroadcastCacheUpdatePlugin` but I can honestly see why it is that way.

If anyway has any alternatives - please suggest

File: packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js [hunk 2]
```diff
@@ -27,47 +27,34 @@ import {defaultHeadersToCheck, defaultSource} from './constants';
  * For efficiency's sake, the underlying response bodies are not compared;
  * only specific response headers are checked.
  *
- * @example <caption>Added as a "plugin" to a `RequestWrapper` to
- * automatically dispatch messages on a cache update</caption>
+ * This class is meant to be agnostic about the service worker environment it's
```
Comment (gauntface) [inline packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js]: Nit: Could you rephrase this to focus on "This class can be used to <insert use here> and can be used in your service worker without any of the other sw-helper modules."

File: packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js [hunk 3]
```diff
@@ -27,47 +27,34 @@ import {defaultHeadersToCheck, defaultSource} from './constants';
...
- *     first: oldResponse,
- *     second: newResponse,
- *     cacheName
- *   });
- * }
+ * Promise.all([
+ *   caches.open(cacheName).then((cache) => cache.match(url)),
+ *   fetch(url),
+ * ]).then(([first, second]) => {
+ *   if (cacheResponse) {
```
Comment (gauntface) [inline packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js]: I dont think cacheResponse is defined

File: packages/sw-broadcast-cache-update/src/index.js (hunk 1 above)
Comment (jeffposnick) [inline packages/sw-broadcast-cache-update/src/index.js]: +1. I didn't know what else to go with, though.

I'll open an issue to discuss possible changes.

File: packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js (hunk 2 above)
Comment (jeffposnick) [inline packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js]: Done.

File: packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js (hunk 3 above)
Comment (jeffposnick) [inline packages/sw-broadcast-cache-update/src/lib/broadcast-cache-update.js]: Righto.
